src/Qwiq.Core/IFieldCollection.cs

```diff
@@ -1,13 +1,15 @@
-using System.Collections;
 using System.Collections.Generic;
 
 namespace Microsoft.Qwiq
 {
-    public interface IFieldCollection : IEnumerable<IField>
+    public interface IFieldCollection : IReadOnlyCollection<IField>
     {
         IField this[string name] { get; }
-        int Count { get; }
-        bool Contains(string fieldName);
+        bool Contains(string name);
+
+        IField TryGetById(int id);
```

Comment:
@pelavall Yeah... 🤷‍♂️ TFS has these weird semantics where it returns null rather than a `bool` and `out` parameter.  Since we're adapting I'm totally fine (:+1:) making it adhere to what a dev would expect from the `Try...` method syntax.